eng/configure-toolset.ps1

```diff
@@ -1,8 +1,13 @@
-# We depend on a local cli for a number of our buildtool
-# commands like init-tools so for now we need to disable
-# using the globally installed dotnet
-
-$script:useInstalledDotNetCli = $false
-
 # Working around issue https://github.com/dotnet/arcade/issues/2673
 $script:DisableNativeToolsetInstalls = $true
+
+function DotnetInstall()
```

Comment:
Why do you need to manually set an environment variable and can't use the one set by InitializeDotNetCli? https://github.com/dotnet/arcade/blob/master/eng/common/tools.ps1#L148. Also that script overrides the path, so just invoking dotnet afterwards should suffice. Why would you even need the `init-dotnet` script? Arcade handles the restore of the toolset. In scripts where you need to invoke msbuild you should probably be using https://github.com/dotnet/arcade/blob/master/eng/common/msbuild.ps1 which handles all that for you. If you need to invoke dotnet/msbuild from inside an msbuild file (proj/targets/prop), you can/should use the arcade defined property: https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/RepoLayout.props#L35-L36.

Additionally to that, the `configure-toolset` script is an arcade entry point which shouldn't contain any functions and shouldn't be imported by our repos.

Reply:
Hmm, for your individual comments:

1) I see the environment getting set in tools.ps1 but it apparently doesn't propagate back to the parent process i.e. from the powershell execution in init-dotnet it doesn't get back to the init-dotnet.cmd script.
2) Hmm, I maybe missing something - how do I call "dotnet.exe" from dotnet.cmd without using init-dotnet to locate / initialize it when it's not yet present?
3) Sounds good, I can easily modify the invocation in tests/build.proj as you suggest.
4) OK, I'll move the script somewhere else unless you explain to me why I don't need it.

Reply:
Fixed in 2nd commit per our offline chat - init-dotnet is deleted and dotnet just uses powershell to call InitializeDotNetCli and uses JanV's trick to capture the return value in an env var that is then used to locate the dotnet tool.